### PR TITLE
Add GitHub workflow for benchmarking

### PR DESCRIPTION
**Pull request by dafeda**
## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kvashchuka** [comment]
Really cool! Is this the correct address for the github page with benchmark results `https://equinor.github.io/ert/`? For some reason the page is blank, maybe I am doing something wrong?🧐

**Comment by dafeda** [comment]
> Really cool! Is this the correct address for the github page with benchmark results `https://equinor.github.io/ert/`? For some reason the page is blank, maybe I am doing something wrong?🧐

The GitHub action I'm using pushes automatically to the [`gh-pages` branch of `ert`](https://github.com/equinor/ert/tree/gh-pages) every time someone commits to `main`.
It pushes to the file `data.js` in the folder `/dev/bench` so the correct url is https://equinor.github.io/ert/dev/bench/

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Benchmark Adaptive Localization
+on:
+  push:
+    branches:
+      - benchmark
+
```
Comment (frode-aarstad) [inline .github/workflows/benchmark.yml:6]: So this benchmark branch should be long lived and when we want to benchmark something we push to it?

Comment (dafeda) [reply .github/workflows/benchmark.yml:6]: I'll need to change this to `main` before merging.

**Comment by frode-aarstad** [comment]
I like it. How could we make sure people know about this? Add a link to github starting page, mention it in the README ...

**Comment by dafeda** [comment]
> I like it. How could we make sure people know about this? Add a link to github starting page, mention it in the README ...

Perhaps we could show it on the screen in the office?

**Review by sondreso — APPROVED** [review]

**Pull request merged**